user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/EditBatchDialog.java

```diff
@@ -0,0 +1,258 @@
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.textfield.TextField;
+import java.util.List;
+import life.qbic.datamanager.views.general.DialogWindow;
+import life.qbic.datamanager.views.projects.project.samples.registration.batch.EditBatchDialog.ConfirmEvent.Data;
+import life.qbic.datamanager.views.projects.project.samples.registration.batch.SampleBatchInformationSpreadsheet.SampleInfo;
+import life.qbic.projectmanagement.domain.model.batch.BatchId;
+import life.qbic.projectmanagement.domain.model.experiment.ExperimentalGroup;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Analyte;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Species;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Specimen;
+/**
+ * A dialog used for editing sample and batch information within an experiment
+ */
+public class EditBatchDialog extends DialogWindow {
+
+  private final SampleBatchInformationSpreadsheet spreadsheet;
+  private final BatchId batchId;
+  private final List<SampleInfo> existingSamples;
+
+  private final TextField batchNameField;
+  private final Span batchNameText;
+
+  public EditBatchDialog(String experimentName,
+      List<Species> species,
+      List<Specimen> specimen,
+      List<Analyte> analytes,
+      List<ExperimentalGroup> experimentalGroups,
+      BatchId batchId,
+      String batchName,
+      List<SampleInfo> existingSamples) {
+
+    addClassName("batch-update-dialog");
+    setConfirmButtonLabel("Update Samples");
+
+    this.batchId = batchId;
+    this.existingSamples = existingSamples.stream().map(SampleInfo::clone).toList();
+
+    spreadsheet = new SampleBatchInformationSpreadsheet(experimentalGroups, species, specimen,
+        analytes, false);
+
+    batchNameField = createBatchNameField();
+    batchNameField.addClassName("batch-name-field");
+    batchNameField.setValue(batchName);
+    batchNameField.addValueChangeListener(this::onBatchNameChanged);
+
+    Button addRow = new Button();
+    addRow.setText("Add Row");
+    addRow.addClickListener(this::onAddRowClicked);
+    addRow.addClassName("add-batch-row");
+
+    Button removeLastRow = new Button();
+    removeLastRow.setText("Remove Row");
+    removeLastRow.addClickListener(this::onRemoveLastRowClicked);
+    removeLastRow.addClassName("remove-batch-row");
+
+    setHeaderTitle("Update Batch");
+    setResizable(true);
+
+    Div batchControls = new Div();
+    batchControls.addClassName("batch-controls");
+    batchControls.add(batchNameField);
+
+    Span userHelp1 = new Span("You are editing samples for batch");
+    Span experimentNameText = new Span(experimentName);
+    experimentNameText.addClassName("experiment-name");
+    Span userHelp2 = new Span("and experiment");
```

Comment:
![Bildschirmfoto 2023-11-24 um 17 26 04](https://github.com/qbicsoftware/data-manager-app/assets/29627977/c778e633-b9a2-48bd-8c01-099bd8deae10)

Minor spacing issue:

```suggestion
    Span userHelp2 = new Span(" and experiment");
```

Reply:
nope the spacing comes from the included component ;) margin-right